sys/sys/memrange.h

```diff
@@ -69,6 +71,16 @@ struct mem_livedump_arg {
 
 #define	MEM_KERNELDUMP	_IOW('m', 53, struct mem_livedump_arg)
 
+#if __has_feature(capabilities)
+struct mem_cheri_cap_arg {
+	ptraddr_t	vaddr;
+	void * __kerncap buf;
+	size_t		len;
+};
+
+#define	MEM_READ_CHERI_CAP	_IOW('m', 54, struct mem_cheri_cap_arg)
```

Comment:
Do we want to leave space to guard against collisions in future merges?

Reply:
Hummm.  I didn't think that would be a problem initially, but it is true that the MEM_EXTRACT_PADDR was added in 2020 and the one for livedumps added in 2022.  The older ones here were added in 1999.